cmd/query/app/flags.go

```diff
@@ -93,6 +99,19 @@ func (qOpts *QueryOptions) InitFromViper(v *viper.Viper, logger *zap.Logger) *Qu
 	return qOpts
 }
 
+// Utility function to get listening address based on port (deprecated flags) or host:port (new flags)
+func getAddressFromCLIOptions(port int, hostPort string) string {
```

Comment:
This function already exists in `./cmd/collector/app/builder_flags.go`, please move it to `./ports/ports.go` as a shared util function `HostPortFromPortOrString(port int, hostPort string)`

Reply:
refactor as suggested